web_src/js/features/repo-settings.js

```diff
@@ -82,6 +82,13 @@ export function initRepoSettingBranches() {
     const $target = $($(this).attr('data-target'));
     if (this.checked) $target.addClass('disabled'); // only disable, do not auto enable
   });
+  $('#dismiss_stale_approvals').on('change', function () {
+    const $target = $('#ignore_stale_approvals_box');
+    $target.toggleClass('disabled', this.checked);
+    if (this.checked) {
+      $('#ignore_stale_approvals').prop('checked', false);
+    }
```

Comment:
```suggestion
```
No need to, better not to interfere with what the user entered.
It is simply meant to show the user that this checkbox is now irrelevant, not to override their input.